Ninja build fails, latest 3.4.1 release

Using Fedora, all deps met. Building from source. 

rpm -q gcc
gcc-10.1.1-2.fc33.x86_64

cmake -H. -Bbuild -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_INSTALL_PREFIX:PATH=~/wz/install -GNinja
cmake --build build --target install

[212/318] Building CXX object src/CMakeFiles/warzone2100.dir/clparse.cpp.o
FAILED: src/CMakeFiles/warzone2100.dir/clparse.cpp.o 
/usr/bin/c++  -DCURL_OPENSSL_DOES_NOT_REQUIRE_LOCKS_INIT -DENABLE_DISCORD -DHAVE_CONFIG_H -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NO_DEBUG -DQT_SCRIPT_LIB -DQT_WIDGETS_LIB -DYY_NO_UNISTD_H -I. -I../. -I../3rdparty -I../3rdparty/date/include -I../3rdparty/discord-rpc/include -I../3rdparty/launchinfo/include -I../3rdparty/EmbeddedJSONSignature/include -I../3rdparty/re2 -isystem ../3rdparty/glm -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtCore -isystem /usr/lib64/qt5/mkspecs/linux-g++ -isystem /usr/include/qt5/QtScript -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtWidgets -fstack-protector-strong  -Wpedantic -Wall -Wextra -fno-common -fno-math-errno -Wcast-align -Wcast-qual -Wctor-dtor-privacy -Winit-self -Woverloaded-virtual -Wstrict-null-sentinel -Wwrite-strings -Wpointer-arith -Wmissing-noreturn -Wnon-virtual-dtor -Wnull-dereference -Wduplicated-cond -Walloc-zero -Walloca-larger-than=1024 -Wrestrict -Wfloat-conversion -Wstringop-truncation -Werror -Wno-logical-op -Wno-float-conversion -Wno-sign-compare -Wno-unused-parameter -Wno-format-truncation -O2 -g -DNDEBUG -fPIE   -fPIC -std=gnu++11 -MD -MT src/CMakeFiles/warzone2100.dir/clparse.cpp.o -MF src/CMakeFiles/warzone2100.dir/clparse.cpp.o.d -o src/CMakeFiles/warzone2100.dir/clparse.cpp.o -c ../src/clparse.cpp
In function 'void poptPrintHelp(poptContext, FILE*)',
    inlined from 'bool ParseCommandLineEarly(int, const char* const*)' at ../src/clparse.cpp:403:17:
../src/clparse.cpp:101:33: error: argument 1 is null but the corresponding size argument 3 range is [1, 128] [-Werror=nonnull]
  101 |   int txtOffset = (int) mbstowcs(nullptr, txt, txtSize) + 1 - txtSize;
      |                         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/10/cstdlib:75,
                 from /usr/include/c++/10/stdlib.h:36,
                 from .././lib/framework/frame.h:27,
                 from ../src/clparse.cpp:27:
/usr/include/stdlib.h: In function 'bool ParseCommandLineEarly(int, const char* const*)':
/usr/include/stdlib.h:933:15: note: in a call to function 'size_t mbstowcs(wchar_t*, const char*, size_t)' declared with attribute 'write_only (1, 3)'
  933 | extern size_t mbstowcs (wchar_t *__restrict  __pwcs,
      |               ^~~~~~~~
cc1plus: all warnings being treated as errors
[217/318] Building CXX object src/CMakeFiles/warzone2100.dir/activity.cpp.o
ninja: build stopped: subcommand failed.

--- a/src/clparse.cpp
+++ b/src/clparse.cpp
@@ -43,6 +43,8 @@
 #include "warzoneconfig.h"
 #include "wrappers.h"
 
+#include <cwchar>
+
 //////
 // Our fine replacement for the popt abomination follows
 
@@ -98,7 +100,9 @@ static void poptPrintHelp(poptContext ctx, FILE *output)
 		// calculate number of terminal columns required to print
 		// for languages with multibyte characters
 		const size_t txtSize = strlen(txt) + 1;
-		int txtOffset = (int) mbstowcs(nullptr, txt, txtSize) + 1 - txtSize;
+		std::mbstate_t state = std::mbstate_t();
+		const char *pTxt = txt;
+		int txtOffset = (int) std::mbsrtowcs(nullptr, &pTxt, 0, &state) + 1 - txtSize;
 		// CJK characters take up two columns
 		char language[3]; // stores ISO 639-1 code
 		strlcpy(language, setlocale(LC_MESSAGES, nullptr), sizeof(language));

